compiler/rustc_mir_transform/src/coroutine/by_move_body.rs

```diff
@@ -134,6 +134,14 @@ pub(crate) fn coroutine_by_move_body_def_id<'tcx>(
             let mut child_precise_captures =
                 child_capture.place.projections[parent_capture.place.projections.len()..].to_vec();
 
+            // Skip trivial captures. These don't need to be remapped, and we save some
```

Comment:
This is both a micro-optimization and also a fix for the case where we have a fake borrow like:

```
_2 = &fake shallow (_1.0: &Foo);
```

and both the parent and child capture `Foo` by ref. In this case, we'll both be inserting a deref (on line 147) but also peeling a deref (on line 158). However, since that fake borrow has no deref to peel, then we would ICE.

Rather, we skip over this capture since there's nothing to actually change here in the body.